core/src/test/scala/org/apache/spark/crypto/OpensslAesCtrCryptoCodecSuite.scala

```diff
@@ -0,0 +1,73 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.spark.crypto
+
+import java.io.{ByteArrayInputStream, BufferedOutputStream, ByteArrayOutputStream}
+import java.security.SecureRandom
+import java.util
+
+import com.intel.chimera.{CryptoInputStream, CryptoOutputStream}
+import com.intel.chimera.codec.OpensslAesCtrCryptoCodec
+import org.apache.spark.{SparkFunSuite, SparkConf}
+import org.apache.spark.crypto.CommonConfigurationKeys._
+import org.scalatest.prop.PropertyChecks
+
+class OpensslAesCtrCryptoCodecSuite extends SparkFunSuite with PropertyChecks  {
```

Comment:
Same thing here. The only thing that comes close to testing Spark code is calling `CryptoStreamUtils.getChimeraProps`, but instead you should write an explicit unit test for that.